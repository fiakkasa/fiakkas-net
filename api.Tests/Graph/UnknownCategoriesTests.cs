using HotChocolate.Execution;

namespace api.Tests;

public class UnknownCategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task UnknownCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  unknownCategories {
    totalCount
    nodes {
      createdAt
      id
      internalId
      title
      updatedAt
      version
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task UnknownCategoryById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  unknownCategoryById(id: "VW5rbm93bkNhdGVnb3J5CmdjOWY1ODc5ZDQwMTg0OWEwOWI3MWI0NzlkZDVkZTdmZg==") {
    createdAt
    id
    internalId
    title
    updatedAt
    version
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}

