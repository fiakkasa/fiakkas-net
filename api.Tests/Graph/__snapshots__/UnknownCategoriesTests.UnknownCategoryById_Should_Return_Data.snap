{
  "data": {
    "unknownCategoryById": {
      "createdAt": "2024-01-01T00:00:00.000Z",
      "id": "VW5rbm93bkNhdGVnb3J5Op2H9ckYQKBJm3G0ed1d5/8=",
      "internalId": "c9f5879d-4018-49a0-9b71-b479dd5de7ff",
      "title": "Title",
      "updatedAt": null,
      "version": 1
    }
  }
}
